src/borg/testsuite/repository.py

```diff
@@ -19,24 +19,28 @@
 
 
 @pytest.fixture()
-def repository(tmpdir):
-    repository_location = os.path.join(str(tmpdir), "repository")
+def repository(tmp_path):
+    repository_location = os.fspath(tmp_path / "repository")
     repository = Repository(repository_location, exclusive=True, create=True)
     yield repository
 
 
 @pytest.fixture()
-def remote_repository(tmpdir):
+def remote_repository(tmp_path):
     if is_win32:
         pytest.skip("On Windows, we skip the remote tests.")
-    repo_location = Location("ssh://__testsuite__" + os.path.join(str(tmpdir), "repository"))
+    repo_location = Location("ssh://__testsuite__" + os.fspath(tmp_path / "repository"))
     return RemoteRepository(repo_location, exclusive=True, create=True)
 
 
 def reopen(repository, exclusive: Optional[bool] = True, create=False):
     if isinstance(repository, Repository):
+        if repository.io or repository.lock is not None:  # The repo should be closed before reopening
+            raise RuntimeError("Cannot support nested repository contexts.")
         return Repository(repository.path, exclusive=exclusive, create=create)
     elif isinstance(repository, RemoteRepository):
+        if repository.p or repository.sock is not None:  # The remote repo should be closed before reopening
+            raise RuntimeError("Cannot support nested repository contexts.")
```

Comment:
In Repository, `close()` sets `io` and `lock` to `None`, so if either of these aren't `None` then the repo should not be reopened.

In RemoteRepository, `close` sets `p` and `sock` to None, so same deal here.

Reply:
You need to check `.io` and `.p` using `is not None`.

Reply:
On it 👍 Thank you.